fullcalendar-app/src/components/FullCalendar.vue

```diff
@@ -172,41 +175,73 @@ export default {
         this.calendarOptions.snapDuration = window.drupalSettings.fullCalendar.snapDuration;
         this.calendarOptions.slotLabelInterval = window.drupalSettings.fullCalendar.slotLabelInterval;
+
+        // Also update minTime and maxTime
+        this.calendarOptions.slotMinTime = window.drupalSettings.fullCalendar.minTime || '05:00:00';
+        this.calendarOptions.slotMaxTime = window.drupalSettings.fullCalendar.maxTime || '23:00:00';
       }
     });
   },
   methods: {
     downloadPDF() {
-      // We are forced to clone the calendar in order to modify its elements
-      // before transferring to the PDF, otherwise, the user will see changes
-      // on the page when the PDF is loaded.
-      const element = document.getElementById('fullcalendar-app');
-      // Create a deep copy of an element.
-      const clone = element.cloneNode(true);
-
-      // Get the text from the calendar header.
-      const scheduleTitle = document.querySelector('.fc-toolbar-title').textContent;
-      const fileName = `Weekly schedule (${scheduleTitle}).pdf`;
+      const calendarClone = this.prepareCalendarClone();
+      this.addCloneToDOM(calendarClone);
 
-      // Setting styles for copy.
+      this.generatePDF(calendarClone).then(({ pdf, filename }) => {
+        pdf.save(filename);
+      }).catch(this.handlePDFGenerationError).finally(() => {
+        this.cleanUp(calendarClone);
+      });
+    },
+    prepareCalendarClone() {
+      const originalCalendar = document.getElementById('fullcalendar-app');
+      const calendarClone = originalCalendar.cloneNode(true);
+
+      this.applyCustomStyles(calendarClone);
+      this.adjustCalendarLayout(calendarClone);
+      this.cleanupUnwantedElements(calendarClone);
+
+      return calendarClone;
+    },
+    applyCustomStyles(clone) {
       clone.style.padding = '0';
       clone.style.marginTop = '10px';
       clone.style.marginLeft = '25px';
       clone.style.marginRight = '25px';
+    },
+    adjustCalendarLayout(clone) {
+      let harness = clone.querySelectorAll('.fc-view-harness.fc-view-harness-active')[0];
+      if (harness) {
+        harness.style.minHeight = '1000px';
+        harness.style.maxHeight = '1020px';
+      }
 
       let tables = clone.querySelectorAll('.fc-timeGridWeek-view table');
       if (tables.length) {
         tables.forEach((table) => {
+          // Apply fixed width for the cloned calendar to standardize layout
           table.style.maxWidth = '1488px';
+          table.style.minWidth = '1488px';
+        });
+      }
+
+      let headerCells = clone.querySelectorAll('#fullcalendar-app th.fc-col-header-cell, #fullcalendar-app th');
+      if (headerCells.length) {
+        headerCells.forEach((cell) => {
+          cell.style.padding = '0';
+          cell.style.height = '1rem';
         });
       }
+      let tableColHeader = clone.querySelectorAll('table.fc-col-header')[0];
+      if (tableColHeader) {
+        tableColHeader.style.margin = '0';
```

Comment:
Same, looks like css class is needed instead of hard coding

Reply:
Fixed.